### ISSUE TITLE
A "container name must be specified" error when deploying a service with hot reload enabled

### ISSUE THREAD

**Issue by eysi09** [garden-io/garden#598]
### Current Behavior

I intermittently get the following error when running `garden dev --hot=go-service`:

```sh
Failed deploying service go-service (from module go-service). Here is the output:
————————————————————————————————————————————————————————————————————————————————
Got error from Kubernetes API - a container name must be specified for pod go-service-568c4bf5cd-9xpps, choose one of: [go-service garden-rsync] or one of the init containers: [garden-sync-init]
———————————————————————————————————————————————————————————————————————————————
```

### Expected behavior

Deployment shouldn't fail.

### Reproducible example

N/A

### Suggested solution(s)

N/A

### Additional context

N/A

### Your environment

`garden version`
Latest `master`

`kubectl version`
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:04:45Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T20:56:12Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by edvald** [comment]
We thought this was fixed in #595, can you check on latest master?

**Comment by eysi09** [comment]
Yes, looks like it's already fixed.
